src/rapidquilt/apply/common.rs

```diff
@@ -112,7 +112,12 @@ pub fn save_modified_file<'arena, H: BuildHasher>(
     }
 
     let file_path = config.base_dir.join(&filename);
+    let mut old_permissions = None;
     if file.existed {
+        if file.permissions.is_none() {
+            let meta = std::fs::metadata(&file_path);
+            old_permissions = Some(meta?.permissions());
```

Comment:
It would be more readable to move the error propagation to the `let` statement, i.e.:
```
let meta = std::fs::metadata(&file_path)?;
old_permissions = Some(meta.permissions());
```